### PR TITLE
LPS-156383: Expose ERC in Structured Content Folders for Asset Libraries

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new?**
- Expose **External Reference Code** in `StructuredContentFolder`.
- Endpoint implementation of `GET`, `UPSERT` and `DELETE` by **External Reference Code** using **AssetLibraryId**.
- Including some tests in API part.

## API
In the API section, this PR contains the new endpoints to operate using the ERC. 

image

To see this new endpoint you have to follow these steps:

1. Deploy `headless-delivery-api` and `headless-delivery-impl` in this order.
2. Go to `/o/api`
3. Go to `StructuredContentFolder`

**JIRA Ticket:**
https://issues.liferay.com/browse/LPS-156383

### CODE REVIEW

**Review by javierdearcos — APPROVED** [review]
Very nice 👏

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentFolderResourceImpl.java [hunk 1]
```diff
@@ -77,6 +77,19 @@
...
 	extends BaseStructuredContentFolderResourceImpl {
 
+	@Override
+	public void
+			deleteAssetLibraryStructuredContentFolderByExternalReferenceCode(
+				Long assetLibraryId, String externalReferenceCode)
+		throws Exception {
+
+		JournalFolder journalFolder =
+			_journalFolderService.getFolderByExternalReferenceCode(
```
Comment (ccorreagg) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentFolderResourceImpl.java:87]: You are not using the created private method

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentFolderResourceImpl.java [hunk 2]
```diff
@@ -402,6 +448,14 @@ private Page<StructuredContentFolder> _getFoldersPage(
 					GetterUtil.getLong(document.get(Field.ENTRY_CLASS_PK)))));
 	}
 
+	private JournalFolder _getStructuredContentFolderByExternalReferenceCode(
```
Comment (ccorreagg) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentFolderResourceImpl.java:451]: To be honest, I don't know if makes sense to have a private method with only one instruction

**Review by ccorreagg — CHANGES_REQUESTED** [review]
Nice work! Just a little comment

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentFolderResourceImpl.java (hunk 2 above)
Comment (javierdearcos) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/StructuredContentFolderResourceImpl.java:451]: Good catch. Better to inline this where it is used

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Review by javierdearcos — APPROVED** [review]

**Review by ccorreagg — APPROVED** [review]

**Comment by ccorreagg** [comment]
ci:forward

**Comment by ccorreagg** [comment]
ci:forward

**Pull request closed without merging**